### PR TITLE
configure: mask password input and default display

### PR DESCRIPTION
**Pull request by javacruft**
Ensure password information is no leaked to the console.

### CODE REVIEW

**Comment by javacruft** [comment]
Tested this manually as wanted to capture the experience:

```
Local or remote access to VMs [local/remote] (local):
CIDR of OpenStack external network - arbitrary but must not be in use (10.20.20.0/24):
Populate OpenStack cloud with demo user, default images, flavors etc [y/n] (y):
Username to use for access to OpenStack (demo):
Password to use for access to OpenStack (mL**********):
Network range to use for project network (192.168.122.0/24):
Setup security group rules for SSH and ICMP ingress [y/n] (y):
Start of IP allocation range for external network (10.20.20.2):
End of IP allocation range for external network (10.20.20.254):
Network type for access to external network [flat/vlan] (flat):
```

File: sunbeam/jobs/questions.py [hunk 1]
```diff
@@ -20,14 +20,30 @@
...
+    def render_default(self, default: DefaultType) -> Text:
+        """Turn the supplied default in to a Text instance.
+
+        Args:
+            default (DefaultType): Default value.
+
+        Returns:
+            Text: Text containing rendering of masked password value.
+        """
+        mask = "*" * (len(default) - 2)
```
Comment (gboutry) [inline sunbeam/jobs/questions.py:43]: Might be a dumb remark, but what if default is either 0 or 1 length ? Do we prevent the password to be 1 in size ?

Comment (javacruft) [reply sunbeam/jobs/questions.py:43]: That's a good point

Comment (wolsen) [reply sunbeam/jobs/questions.py:43]: Hmm this is still leaky. The supplied algorithm reveals some chars and the length of the password. I think it's better to just dump a consistent x number of asterisks.

Comment (javacruft) [reply sunbeam/jobs/questions.py]: OK I've updated to use a fixed size mask so that the password length is not actually displayed

**Comment by gboutry** [comment]
If running with the "-v" flag, you get:
```
[13:29:48] DEBUG    Value from default function q5P4CK9LX9ST                                                                                                                                                                            questions.py:101
Password to use for access to OpenStack (q5********): 
```
Still showing the password on screen

**Review by gnuoy — APPROVED** [review]

**Pull request merged**